iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -636,10 +641,12 @@ private async Task VerifyIsOpenAsync(CancellationToken cancellationToken)
             }
         }
 
-        private async Task OpenInternalAsync(CancellationToken cancellationToken)
+        private async Task OpenInternalAsync(bool recoverSubscriptions, CancellationToken cancellationToken)
         {
             var connectionStatusInfo = new ConnectionStatusInfo();
 
+            _handleDisconnectCts ??= new CancellationTokenSource();
```

Comment:
The cancellation token guiding reconnection attempts is initialized when the client is opened. This cancellation token source is canceled, disposed and set to null when the client is either closed, it is disposed or the connection attempt fails.